### PR TITLE
Add a multifile partial clutz test.

### PR DESCRIPTION
**Pull request by LucasSloan**
Adds a test case for compiling 3 javascript files (1 importing from the other 2) individually, then running a DeclarationSyntaxTest on the result to ensure that Clutz produces correct output.

As a result, an error in ImportRenameMapBuilder was discovered where the imported symbols were being rename to modulename *underscore* symbolname, instead of the correct modulename *dot* symbolname.

### CODE REVIEW

File: src/test/java/com/google/javascript/clutz/DeclarationGeneratorTests.java [hunk 1]
```diff
@@ -97,8 +97,11 @@ static String getExternFileNameOrNull(String testFileName) {
     File[] testFiles = getPackagePath().toFile().listFiles(filter);
     // Partial files live in 'partial' dir and run implicitly with the --partialInput option on.
     File[] testPartialFiles = getPackagePath().resolve("partial").toFile().listFiles(filter);
+    File[] testMultifilePartailFiles =
```
Comment (rkirov) [inline src/test/java/com/google/javascript/clutz/DeclarationGeneratorTests.java]: Add a comment describing the purpose of multifile partial, similar to the one above. Call out specifically that each .js creates its own golden test *and* we do a singleton test on top.

File: src/test/java/com/google/javascript/clutz/MultiFileTest.java [hunk 2]
```diff
@@ -83,6 +83,15 @@ public void skipEmitWithTypedefs() throws Exception {
         .generatesDeclarations(expected);
   }
 
+  @Test
+  public void multifilePartial() throws Exception {
+    String expected = DeclarationGeneratorTests.getTestFileText(input("singleton.d.ts"));
```
Comment (rkirov) [inline src/test/java/com/google/javascript/clutz/MultiFileTest.java]: nitpicing on the name: 'singleton' in a java code base might bring unwanted associations. How about 'total', that's my new word for old clutz mode.

File: src/test/java/com/google/javascript/clutz/multifilePartial/singleton.d.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+//!! The singleton file is the result of a global compilation of the files in
```
Comment (rkirov) [inline src/test/java/com/google/javascript/clutz/multifilePartial/singleton.d.ts]: global -> total

File: src/test/java/com/google/javascript/clutz/multifilePartial/singleton.d.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+//!! the multifilePartial directory, so that the global and partial compilations
+//!! can be compared
```
Comment (rkirov) [inline src/test/java/com/google/javascript/clutz/multifilePartial/singleton.d.ts]: ok for now, but long term this file will be less and less useful (if we remove total mode one day). An even better test (in another PR) would a usage.ts file that typechecks together with the three partial .d.ts.

File: src/main/java/com/google/javascript/clutz/ImportRenameMapBuilder.java [hunk 5]
```diff
@@ -152,7 +152,7 @@ private static String buildWholeModuleExportSymbolName(String importedModuleId)
    * https://github.com/angular/clutz/issues/596
    */
   private static String buildNamedExportSymbolName(String importedModuleId, String originalName) {
-    return "module$exports$" + importedModuleId.replace(".", "$") + "_" + originalName;
+    return "module$exports$" + importedModuleId.replace(".", "$") + "." + originalName;
```
Comment (rkirov) [inline src/main/java/com/google/javascript/clutz/ImportRenameMapBuilder.java]: does this need a change in tsickle?

Comment (LucasSloan) [reply src/main/java/com/google/javascript/clutz/ImportRenameMapBuilder.java]: Yes - https://github.com/angular/tsickle/pull/671

File: src/test/java/com/google/javascript/clutz/DeclarationGeneratorTests.java (hunk 1 above)
Comment (LucasSloan) [reply src/test/java/com/google/javascript/clutz/DeclarationGeneratorTests.java]: Added!

File: src/test/java/com/google/javascript/clutz/MultiFileTest.java (hunk 2 above)
Comment (LucasSloan) [reply src/test/java/com/google/javascript/clutz/MultiFileTest.java]: Huh - what does singleton mean in java?

And changed.

File: src/test/java/com/google/javascript/clutz/multifilePartial/singleton.d.ts (hunk 3 above)
Comment (LucasSloan) [reply src/test/java/com/google/javascript/clutz/multifilePartial/singleton.d.ts]: Done.

File: src/test/java/com/google/javascript/clutz/multifilePartial/singleton.d.ts (hunk 4 above)
Comment (LucasSloan) [reply src/test/java/com/google/javascript/clutz/multifilePartial/singleton.d.ts]: That makes sense!

**Comment by LucasSloan** [comment]
PTAL.

**Pull request merged**